Test feature survey submission

JoeDKoal94 opened the pull request:
Tests feature for surveySubmission

yarashlee committed:
added test

marcjacobroy committed:
made formresponsedto model

marcjacobroy committed:
create associatesurveysession model

marcjacobroy committed:
add controller/service/repo for associatesurveysession

leonardoz15 committed:
added tests for AssociateSurveySessionController, needs docs

leonardoz15 committed:
finished tests for createAssociateSurveySession service method

leonardoz15 committed:
test for getAssociateSurveySession in controller

leonardoz15 committed:
test for readAssociateSurveySession in service

leonardoz15 committed:
completed tests for updateAssociateSurveySession in controller

leonardoz15 committed:
completed test for updateAssociateSurveySession in service

leonardoz15 committed:
Updated associateSurveySession controller endpoint names

marcjacobroy committed:
implement associatesurveyservice controller

marcjacobroy committed:
implement associateSurveyService

marcjacobroy committed:
small style change in associatesurveyservice

leonardoz15 committed:
working controller tests

changed post endpoint to accept a AssociateSurveySubmissionDto as a requestBody, created this Dto object, and refactored tests

leonardoz15 committed:
finished service tests for associateSurveyService

leonardoz15 committed:
uncommented first create service test

marcjacobroy committed:
update tests/add associate survey service dto

marcjacobroy committed:
finish documentation of test code

leonardoz15 committed:
final documentation for AssociateSurveySession tests

leonardoz15 committed:
Delete sync-errors.html

jurczakn commented on SyncService/src/main/java/com/revature/dto/AssociateSurveySessionDto.java:
```diff
@@ -0,0 +1,111 @@
+
+public class AssociateSurveySessionDto implements Dto<AssociateSurveySession> {
+
+	int surveyId;
```
Why not private?

jurczakn commented on SyncService/src/main/java/com/revature/controllers/AssociateSurveySessionController.java:
```diff
@@ -0,0 +1,80 @@
+
+/**
+ * 
+ * This controller handles endpoints for the CRUD operations of
```
try to be more descriptive

jurczakn commented on SyncService/src/main/java/com/revature/repo/SurveyRepo.java:
```diff
@@ -0,0 +1,10 @@
+package com.revature.repo;
```
remove

jurczakn commented on SyncService/src/test/java/com/revature/controllers/AssociateSurveySessionControllerTest.java:
```diff
@@ -0,0 +1,227 @@
+	 */
+	@Test
+	void createAssociateSurveySession_withoutErrors() throws JsonProcessingException {
+		when(service.createAssociateSurveySession(Mockito.anyInt(), Mockito.anyInt(), Mockito.anyString()))
```
change to check the actual values

jurczakn commented on SyncService/src/test/java/com/revature/controllers/AssociateSurveySessionControllerTest.java:
```diff
@@ -0,0 +1,227 @@
+	 * status code.
+	 */
+	@Test
+	void updateAssociateSurveySession_nullInput() {
```
how to tell difference from service returning null

jurczakn replied:
probably don't even need to call service if parameter is not part of request

jurczakn commented on SyncService/src/test/java/com/revature/controllers/SurveyControllerTest.java:
```diff
@@ -0,0 +1,108 @@
+	@AfterEach
+	void tearDown() throws Exception {
+	}
+
```
test if doesn't exist

jurczakn commented on SyncService/src/test/java/com/revature/service/AssociateSurveySessionServiceTest.java:
```diff
@@ -0,0 +1,164 @@
+				associateSurveySessionCopy.getAssociateId(), associateSurveySessionCopy.getSurveyId(),
+				associateSurveySessionCopy.getBatchId());
+
+		verify(repo, times(1)).save(returnedFirst);
```
should not call save if object already exists.

jurczakn commented on SyncService/src/test/java/com/revature/service/SurveyServiceTest.java:
```diff
@@ -0,0 +1,82 @@
+	@AfterEach
+	void tearDown() throws Exception {
+	}
+
```
test if survey does not exist

jurczakn replied:
test if cannot reach survey service

leonardoz15 committed:
removed unnecessary files for pull request

removed outdated survey/survey question related files, added @SpringBootTest to AssociateSurveySessionService test, added .DS_Store and sync-errors.html to gitignore

leonardoz15 committed:
removed sync-errors.html

leonardoz15 committed:
updated dto interface to match dev branch

leonardoz15 committed:
updated dto interface to match dev branch

leonardoz15 committed:
refactor of AssociateSurveySession controller

removed create and update methods of associateSurveySession controller, added custom update exception, refactored associateSurveySession service tests to include more edge cases when updating an associateSurveySession

leonardoz15 committed:
Merge branch 'dev' into test-feature-survey-submission

leonardoz15 committed:
added test for failure creating AssociateSurveySession

leonardoz15 committed:
Merge branch 'feature-survey-submission' into test-feature-survey-submission

leonardoz15 committed:
fixed controler test, added documentation for AssociateSurveySession service

leonardoz15 committed:
updated service and controller tests

Signed-off-by: Zachary Leonardo <zachleonardo15@gmail.com>

leonardoz15 committed:
Delete sync-errors.html

leonardoz15 committed:
removed FormResponseDto
